app/views/components/_autocomplete.html.erb

```diff
@@ -1,27 +1,24 @@
 <%
   id ||= "autocomplete-#{SecureRandom.hex(4)}"
-  type ||= nil
   error_id = "error-#{SecureRandom.hex(4)}"
   error_items ||= []
-  hint ||= nil
-  width ||= nil
   aria = error_id if error_items.any?
-  data_attributes ||= {}
   track_category ||= nil
   track_label ||= nil
   select ||= {}
-  input ||= {}
-  search ||= false
-  jsonly ||= false
+  autocomplete_configuration_options ||= {}
+
+  data_attributes ||= {}
+  data_attributes[:module] ||= ""
+  data_attributes[:module] << " autocomplete"
+  data_attributes[:module] = data_attributes[:module].strip!
```

Comment:
_Tiny nitpick_ about the difference between [`String#strip` and `String#strip!`](https://ruby-doc.org/core-3.1.1/String.html#method-i-strip)...

The `strip!` method mutates the existing String object, so you shouldn't need to assign it back to itself.

Whereas `strip` returns a new String object, but doesn't mutate the existing one. And therefore you need to assign it if you want to keep it.

So this could be rewritten as:

```suggestion
  data_attributes[:module].strip!
```

or alternatively:

```suggestion
  data_attributes[:module] = data_attributes[:module].strip
```

We don't need to do both (i.e. re-assign and also mutate the String).